src/operations/blas1/IndexMaxMin.hpp

```diff
@@ -94,22 +94,14 @@ PORTBLAS_INLINE void IndexMaxMin<is_max, is_step0, lhs_t, rhs_t>::eval(
   using element_t =
       typename ResolveReturnType<op, rhs_t>::type::value_t::value_t;
 
-#ifndef __ADAPTIVECPP__
   // reduction within the sub_group
   for (index_t i = sg_local_range >> 1; i > 0; i >>= 1) {
-    if (sg_local_id < i) {
-      element_t shfl_val = sycl::shift_group_left(sg, val.get_value(), i);
-      index_t shfl_idx = sycl::shift_group_left(sg, val.get_index(), i);
-      value_t shfl{shfl_idx, shfl_val};
-      val = op::eval(val, shfl);
-    }
+    element_t shfl_val = cl::sycl::shift_group_left(sg, val.get_value(), i);
+    index_t shfl_idx = cl::sycl::shift_group_left(sg, val.get_index(), i);
+    value_t shfl{shfl_idx, shfl_val};
+    val = op::eval(val, shfl);
```

Comment:
Have you checked that this is not going to read out of bounds and potentially fail the iamax or iamin?

Reply:
I am not sure how it could read out of bounds and in case how to properly check it.
The `val` variable is set for every thread to a neutral value for the operator at the beginning of the kernel. Then for `global_linear_id` smaller than the input array size is iteratively set using the `op::eval`. AFAIK the shift function is inside the subgroup so no random value should be read here. 
The only misleading point I see is that `val` is updated for all thread even if, at the end, only the thread with `sg_local_id` equal to zero has the correct final result. We don't care about it since this is the only thread that will write its result to memory.
If I am missing something on how to check it please let me know.

Reply:
You're right. I had flashbacks of out of bound issues with other group algorithms that were taking pointers so removing a similar `if` condition was not trivial.
As you said this looks fine here, thanks!